src/bson.rs

```diff
@@ -1054,6 +1057,131 @@ impl Binary {
     }
 }
 
+#[cfg(feature = "uuid-0_8")]
+#[cfg_attr(docsrs, doc(cfg(feature = "uuid-0_8")))]
+/// Enum of the possible representations to use when converting between Uuid and Binary.
+/// This enum is necessary because the different drivers used to have different ways of encoding
+/// UUID, with the BSON subtype: 0x03 UUID old.
+/// If a UUID has been serialized with a particular representation, it MUST
+/// be deserialized with the same representation.
+///
+/// Example:
+/// ```
+/// use crate::{bson::UuidRepresentation, bson::Binary};
+/// use uuid::Uuid;
+/// let uuid = Uuid::parse_str("00112233445566778899AABBCCDDEEFF").unwrap();
+/// let bin = Binary::from_uuid_with_representation(uuid, UuidRepresentation::PythonLegacy);
+/// let new_uuid = bin.to_uuid();
+/// assert!(new_uuid.is_err());
+/// let new_uuid = bin.to_uuid_with_representation(UuidRepresentation::PythonLegacy);
+/// assert!(new_uuid.is_ok());
+/// assert_eq!(new_uuid.unwrap(), uuid);
+/// ```
+#[non_exhaustive]
+#[derive(PartialEq, Clone, Copy, Debug)]
+pub enum UuidRepresentation {
+    /// The canonical representation of UUIDs in BSON (binary with subtype 0x04)
+    Standard,
+    /// The legacy representation of UUIDs in BSON used by the C# driver (binary subtype 0x03)
+    CSharpLegacy,
+    /// The legacy representation of UUIDs in BSON used by the Java driver (binary subtype 0x03)
+    JavaLegacy,
+    /// The legacy representation of UUIDs in BSON used by the Python driver, which is the same
+    /// format as STANDARD, but has binary subtype 0x03
+    PythonLegacy,
+}
+
+#[cfg(feature = "uuid-0_8")]
+#[cfg_attr(docsrs, doc(cfg(feature = "uuid-0_8")))]
+impl Binary {
+    /// Serializes a UUID into BSON binary type
+    pub fn from_uuid(uuid: uuid::Uuid) -> Self {
+        Binary::from(uuid)
+    }
+    /// Serializes a UUID into BSON binary type and takes the desired representation as a parameter.
+    /// Binary::from_uuid_with_representation(uuid, UuidRepresentation::Standard) is equivalent
```

Comment:
These docstrings are markdown, so you can enclose the code portions in \`\` for easier reading. Also, in rust docs if you also enclose them in square brackets `[]`, it'll automatically link to the thing. e.g. [\`Binary::from_uuid\`].